lib/matplotlib/tests/test_colors.py

```diff
@@ -2060,3 +2061,168 @@ def test_mult_norm_call_types():
+    data = np.array([[0.0, 0], [1, 1]])
+    mdata = mcolorizer._ensure_multivariate_data(data, 2)
+    assert mdata.shape == (2,)
+    assert mdata.dtype.fields['f0'][0] == np.float64
+    assert mdata.dtype.fields['f1'][0] == np.float64
+
+    # test more input dims
+    data = np.zeros((3, 4, 5, 6))
+    mdata = mcolorizer._ensure_multivariate_data(data, 3)
+    assert mdata.shape == (4, 5, 6)
+
+
+def test_colorizer_multinorm_implicit():
+    ca = mcolorizer.Colorizer('BiOrangeBlue')
+    ca.vmin = (0, 0)
+    ca.vmax = (1, 1)
+
+    # test call with two single values
+    data = [0.1, 0.2]
+    res = (0.10009765625, 0.1510859375, 0.20166015625, 1.0)
+    assert_array_almost_equal(ca.to_rgba(data), res)
+
+    # test call with two 1d arrays
+    data = [[0.1, 0.2], [0.3, 0.4]]
+    res = [[0.10009766, 0.19998877, 0.29931641, 1.],
+           [0.20166016, 0.30098633, 0.40087891, 1.]]
+    assert_array_almost_equal(ca.to_rgba(data), res)
+
+    # test call with two 2d arrays
+    data = [np.linspace(0, 1, 12).reshape(3, 4),
+            np.linspace(1, 0, 12).reshape(3, 4)]
+    res = np.array([[[0.00244141, 0.50048437, 0.99853516, 1.],
+                     [0.09228516, 0.50048437, 0.90869141, 1.],
+                     [0.18212891, 0.50048437, 0.81884766, 1.],
+                     [0.27197266, 0.50048437, 0.72900391, 1.]],
+                    [[0.36572266, 0.50048437, 0.63525391, 1.],
+                     [0.45556641, 0.50048438, 0.54541016, 1.],
+                     [0.54541016, 0.50048438, 0.45556641, 1.],
+                     [0.63525391, 0.50048437, 0.36572266, 1.]],
+                    [[0.72900391, 0.50048437, 0.27197266, 1.],
+                     [0.81884766, 0.50048437, 0.18212891, 1.],
+                     [0.90869141, 0.50048437, 0.09228516, 1.],
+                     [0.99853516, 0.50048437, 0.00244141, 1.]]])
+    assert_array_almost_equal(ca.to_rgba(data), res)
+
+    with pytest.raises(ValueError):
+        ca.to_rgba([[0.1, 0.2, 0.3]])
+    with pytest.raises(ValueError):
+        ca.to_rgba([[0.1]])
+
+    # test multivariate
+    ca = mcolorizer.Colorizer('3VarAddA')
+    ca.vmin = (-0.1, -0.2, -0.3)
+    ca.vmax = (0.1, 0.2, 0.3)
+
+    data = [0.1, 0.1, 0.1]
+    res = (0.712612, 0.896847, 0.954494, 1.0)
+    assert_array_almost_equal(ca.to_rgba(data), res)
+
+
+def test_colorizer_multinorm_explicit():
+
+    with pytest.raises(ValueError, match="MultiNorm must be assigned"):
+        ca = mcolorizer.Colorizer('BiOrangeBlue', 'linear')
+
+    with pytest.raises(TypeError,
+                       match=("'norm' must be an instance of matplotlib.colors.Norm"
+                              ", str or None, not a list")):
+        ca = mcolorizer.Colorizer('viridis', ['linear', 'linear'])
+
+    with pytest.raises(ValueError,
+                       match=("Invalid norm for multivariate colormap with 2 inputs")):
+        ca = mcolorizer.Colorizer('BiOrangeBlue', ['linear', 'linear', "log"])
```

Comment:
Be consistent with quotes at least in a line.
```suggestion
        ca = mcolorizer.Colorizer('BiOrangeBlue', ['linear', 'linear', 'log'])
```